WalletWasabi/Blockchain/TransactionOutputs/CoinsRegistry.cs

```diff
@@ -198,6 +206,11 @@ private bool TryGetSpenderSmartCoinsByOutPointNoLock(OutPoint outPoint, [NotNull
 		return CoinsByOutPoint.TryGetValue(outPoint, out coins);
 	}
 
+	public bool IsCoinBanned(SmartCoin coin, DateTimeOffset when, [NotNullWhen(true)] out DateTimeOffset? bannedUntil)
+	{
+		return CoinPrison.IsCoinBanned(coin, when, out bannedUntil);
```

Comment:
A coin that is not in the registry cannot be banned or unbanned. CoinRegistry is not a pass-through class, it is its responsibility to verify all queries about coins make sense.

Also, in general a method has to do something useful and not return raw data because in that way it is the caller the one that has to do the job. If you return a date then the caller has to make the comparison everywhere so, the method should receive the datetime and return only one boolean indicating whether or not the coins is banned at the specified time. 

In case the caller needs to know the `bannedUntil` then you can have a method like this one that follows the `bool TryGetXXX(theThing, out XXX? theXXX)` .